app/gui/src/providers/auth.ts

```diff
@@ -52,6 +53,12 @@ export function createUsersMeQuery(
         void setUsername(sessionVal.email)
         return null
       }
+      if (user.plan === backendModule.Plan.free && refetchCount.value < 30) {
+        return null
```

Comment:
i would probably wait for a timeout here `await new Promise(resolve => { setTimeout(resolve, 3_000) })`

Reply:
why is that? we only want to check for `user.plan` for ~30secs. If nothing changes then user should be redirected to the dashboard. Adding timeout here makes this awaiting indefinitely, right?

Reply:
oh nvm, `refetchSession` is called in `Registration.tsx` behind a timeout so nothing needs to be done here